src/Middleware/HttpOverrides/src/IPNetwork.cs

```diff
@@ -83,21 +93,159 @@ private byte[] CreateMask()
+    /// or <see langword="null"/> if the conversion failed. This parameter is passed uninitialized.
+    /// </param>
+    /// <returns>
+    /// <see langword="true"/> if the <paramref name="networkString"/> parameter was
+    /// converted successfully; otherwise <see langword="false"/>.
+    /// </returns>
+    /// <inheritdoc cref="TryParseComponents(ReadOnlySpan{char}, out IPAddress?, out int)"/>
+    public static bool TryParse([NotNullWhen(true)] string? networkString, [NotNullWhen(true)] out IPNetwork? network)
+    {
+        return TryParse(networkString.AsSpan(), out network);
+    }
+
+    /// <summary>
+    /// Converts the specified <see cref="ReadOnlySpan{T}"/> of <see langword="char"/> representation of
+    /// an IP address and a prefix length to its <see cref="IPNetwork"/> equivalent, and returns a value
+    /// that indicates whether the conversion succeeded.
+    /// </summary>
+    /// <param name="networkSpan">The <see cref="ReadOnlySpan{T}"/> of <see langword="char"/> to validate.</param>
+    /// <param name="network">
+    /// When this method returns, contains the <see cref="IPNetwork"/> equivalent to the IP Address
+    /// and prefix length contained in <paramref name="networkSpan"/>, if the conversion succeeded,
+    /// or <see langword="null"/> if the conversion failed. This parameter is passed uninitialized.
+    /// </param>
+    /// <returns>
+    /// <see langword="true"/> if the <paramref name="networkSpan"/> parameter was
+    /// converted successfully; otherwise <see langword="false"/>.
+    /// </returns>
+    /// <inheritdoc cref="TryParseComponents(ReadOnlySpan{char}, out IPAddress?, out int)"/>
+    public static bool TryParse(ReadOnlySpan<char> networkSpan, [NotNullWhen(true)] out IPNetwork? network)
+    {
+        network = null;
+
+        if (!TryParseComponents(networkSpan, out var prefix, out var prefixLength))
+        {
+            return false;
+        }
+
+        if (!IsValidPrefixLengthRange(prefix, prefixLength))
+        {
+            return false;
+        }
+
+        network = new IPNetwork(prefix, prefixLength, false);
+        return true;
+    }
+
+    /// <remarks>
+    /// <para>
+    /// The specified representation must be expressed using CIDR (Classless Inter-Domain Routing) notation, or 'slash notation',
+    /// which contains an IPv4 or IPv6 address and the subnet mask prefix length, separated by a forward slash.
+    /// </para>
+    /// <example>
+    /// e.g. <c>"192.168.0.1/31"</c> for IPv4, <c>"2001:db8:3c4d::1/127"</c> for IPv6
+    /// </example>
+    /// </remarks>
+    private static bool TryParseComponents(
+        ReadOnlySpan<char> networkSpan,
+        [NotNullWhen(true)] out IPAddress? prefix,
+        out int prefixLength)
+    {
+        prefix = null;
+        prefixLength = default;
+
+        var forwardSlashIndex = networkSpan.IndexOf('/');
+        if (forwardSlashIndex == -1)
```

Comment:
```suggestion
        if (forwardSlashIndex < 0)
```
The comparison with 0 is tiny little bit cheaper than the one to -1, because at machine code it's a `test` instruction (instead of `cmp`) and that can be fused with the jump at machine-level to only one instruction actually executed.
Little effect, but the code change is also little, and it's cheap 😉.

Reply:
Interesting! So I'm assuming this uses a conditional jump with something like `jl` in this case?

Updated!

Reply:
It's still a jump, but CPUs can execute the `test + jmp`-combo more efficiently than `cmp + jmp`. The former will be fused to only one code instruction that the CPU needs to execute (instead of two).